File: boj/1092/main/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
	var crains []int
	var n, m int
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	sc.Scan()
	n, _ = strconv.Atoi(sc.Text())
	for i := 0; i < n; i++ {
		sc.Scan()
		b, _ := strconv.Atoi(sc.Text())
		crains = append(crains, b)
	}

	sc.Scan()
	m, _ = strconv.Atoi(sc.Text())
	var boxes []int
	for i := 0; i < m; i++ {
		sc.Scan()
		b, _ := strconv.Atoi(sc.Text())
		boxes = append(boxes, b)
	}

	sort.Ints(crains)
	crainPos := make([]int, n)
	boxCheck := make([]bool, m)
	for i, j := 0, len(crains)-1; i < j; i, j = i+1, j-1 {
		crains[i], crains[j] = crains[j], crains[i]
	}
	sort.Ints(boxes)

	for i, j := 0, len(boxes)-1; i < j; i, j = i+1, j-1 {
		boxes[i], boxes[j] = boxes[j], boxes[i]
	}
	count := 0

	if boxes[0] > crains[0] {
		fmt.Printf("%d", -1)
		return
	}
	checked := 0
	for checked < m {
		for i := range crains {
			for j := crainPos[i]; j < m; j++ {
				if !boxCheck[j] && crains[i] >= boxes[j] {
					crainPos[i]++
					boxCheck[j] = true
					checked++
					break
				}
				crainPos[i]++
			}
		}
		count++
	}
	fmt.Printf("%d", count)
}

File: boj/1092/main.py
import sys
input = sys.stdin.readline


def solution():
    n = int(input())
    cranes = list(map(int, input().split()))
    m = int(input())
    boxes = list(map(int, input().split()))
    cnt = 0

    cranes.sort(reverse=True)
    boxes.sort(reverse=True)
    crane_pos = [0] * n
    box_check = [True] * m
    checked = 0
    if cranes[0] < boxes[0]:
        print(-1)
        return
    else:
        while checked < m:
            for i in range(n):
                for j in range(crane_pos[i], m):
                    if box_check[j] and cranes[i] >= boxes[j]:
                        crane_pos[i] += 1
                        box_check[j] = False
                        checked += 1
                        break
                    crane_pos[i] += 1
            cnt += 1
        print(cnt)

solution()